File: php/_1_srp_single_responsibility_principle/App/ReportGenerator.php
<?php

namespace SRP\App;

use SRP\App\Student;
use SRP\App\GradeCalculater;

class ReportGenerator
{
    private $calc;

    public function __construct(GradeCalculater $calc)
    {
        $this->calc = $calc;
    }

    public function generate(Student $student): void
    {
        echo "Name {$student->getName()} is {$this->calc->status($student)}. Grade = {$this->calc->average($student)}\n";
    }
}

File: python/_1_srp_single_responsibility_principle/apps/ReportGenerator.py
from .Student import Student
from .GradeCalculater import GradeCalculater


class ReportGenerator:
    def __init__(self, calc: GradeCalculater):
        self.calc = calc
    
    def generate(self, student: Student) -> None:
        average = self.calc.average(student=student)
        status = self.calc.status(student=student)
        print(f'Name: {student.get_name()} is {status}. Grade Avg = {average:.2f}')